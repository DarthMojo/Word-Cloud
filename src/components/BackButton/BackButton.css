.BackButton {
    position: absolute;
    right: 32px;
    top: 32px;
    width: 32px;
    height: 32px;
    opacity: 0.8;
}
.BackButton:hover {
    opacity: 1;
}
.BackButton:before, .BackButton:after {
    position: absolute;
    left: 15px;
    content: ' ';
    height: 33px;
    width: 2px;
    background-color: #333;
}
.BackButton:before {
    transform: rotate(45deg);
}
.BackButton:after {
    transform: rotate(-45deg);
}
