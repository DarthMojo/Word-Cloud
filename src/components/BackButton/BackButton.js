import React from "react";
import type { Element } from "react";

import "./BackButton.css";

type BackButtonType = {
    onButtonClick: Function
}
const BackButton = ({ onButtonClick }: BackButtonType): Element<any> => {
    return (<div>
        <a className={"BackButton"}
            onClick={onButtonClick} />
    </div>);
};

export default BackButton;
