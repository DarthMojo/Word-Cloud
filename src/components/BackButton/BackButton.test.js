/* eslint-disable no-undef */
import React from "react";
import { shallowToJson } from "enzyme-to-json";
import BackButton from "./BackButton";

test("render BackButton()", () => {
    const onButtonClick = () => {};
    const wrapper = shallow(
        <BackButton
            onButtonClick={onButtonClick}
        />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
});
