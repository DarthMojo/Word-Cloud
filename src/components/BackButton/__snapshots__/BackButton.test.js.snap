// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render BackButton() 1`] = `
<div>
  <a
    className="BackButton"
    onClick={[Function]}
  />
</div>
`;
